nav2_smac_planner/src/node_lattice.cpp

```diff
@@ -327,13 +327,18 @@ float NodeLattice::getTraversalCost(const NodePtr & child)
 
 float NodeLattice::getHeuristicCost(
   const Coordinates & node_coords,
-  const Coordinates & goal_coords)
+  const CoordinateVector & goals_coords)
 {
   // get obstacle heuristic value
+  // obstacle heuristic does not depend on goal heading
   const float obstacle_heuristic = getObstacleHeuristic(
-    node_coords, goal_coords, motion_table.cost_penalty);
-  const float distance_heuristic =
-    getDistanceHeuristic(node_coords, goal_coords, obstacle_heuristic);
+    node_coords, goals_coords[0], motion_table.cost_penalty);
+  float distance_heuristic = std::numeric_limits<float>::max();
+  for (unsigned int i = 0; i < goals_coords.size(); i++) {
```

Comment:
I'd be interested in the runtime impact of this over the all headings setting

Reply:
Included in the pr descrition

Reply:
That would be alot of memory, leaving this as a single table is best I think -- but I think this is moot now from the discussion below?

Reply:
I  am just starting to see a lots of problem with the 3 table idea, because if i precompute the table, there is a possibility the minimum distance heuristic corresponding to a goal be prune because now i do goal path pruning. The other option is to create a default table in the beginning during the is path valid function, I can precompute a temporal table based on this default. This way in defautl goal heading mode, I dont influence the computation time, what do you think?. The only downside here is that during I will have two lookup tables in memory though.

Reply:
I don't quite understand what you're asking. 

I think we should leave the table structure as it is. The content should just be set according to the heading policy

Reply:
The problem with setting it beforehand is that there is a change that one of the goals set to be minimum will be pruned later in the isinputvalid function making the lookup table invalid hence why I decided to go with the current approach. I am working on the following, still needs testing and benchmarking but for the last few days, I have been having issues building and running dockers in rolling(the comment from the other issue :).  This is still wip 

```
void NodeHybrid::resetDistanceHeuristic(
  const CoordinateVector & goals_coords,
  const GoalHeadingMode & goal_heading_mode)
{
  dist_heuristic_lookup_table = initial_dist_heuristic_lookup_table;

  if (goal_heading_mode == GoalHeadingMode::DEFAULT) {
    return;
  }

  std::unordered_set<int> goal_heading_angles;
  if (goal_heading_mode == GoalHeadingMode::BIDIRECTIONAL ||
    goal_heading_mode == GoalHeadingMode::ALL_DIRECTION)
  {
    goal_heading_angles.reserve(goals_coords.size());
    for (const auto & coord : goals_coords) {
      goal_heading_angles.insert(motion_table.getClosestAngularBin(coord.theta));
    }
  }
  int ceiling_size = static_cast<int>(ceil(size_lookup / 2.0));
  int total_size = static_cast<int>(size_lookup);
  auto is_goal_heading = [&](int heading) {
      return goal_heading_angles.find(heading) != goal_heading_angles.end();
    };
  int index = 0;
  if (goal_heading_mode == GoalHeadingMode::BIDIRECTIONAL) {
    for (float x = ceil(-size_lookup / 2.0); x <= floor(size_lookup / 2.0); x += 1.0) {
      for (float y = 0.0; y <= floor(size_lookup / 2.0); y += 1.0) {
        for (int heading = 0; heading < dim_3_size_int; ++heading) {
          int heading_angle_180 = (heading + dim_3_size_int / 2) % dim_3_size_int;
          int index_180 = (index-heading) + heading_angle_180;
          // std::cout << "index: " << index << " index_180: " << index_180 << " size: " << dist_heuristic_lookup_table.size() << std::endl;

          float dist_heuristic = initial_dist_heuristic_lookup_table[index];
          float dist_heuristic_180 = initial_dist_heuristic_lookup_table[index_180];

          if (dist_heuristic_180 < dist_heuristic && is_goal_heading(heading_angle_180)) {
            dist_heuristic_lookup_table[index] = dist_heuristic_180;
          }
          index++;
        }
      }
    }
  } else if (goal_heading_mode == GoalHeadingMode::ALL_DIRECTION) {
    for (int x = 0; x < size_lookup; ++x) {
      for (int y = 0; y < ceiling_size; ++y) {
        std::vector<float> dist_heuristic_angles(dim_3_size_int);
        std::vector<int> indices(dim_3_size_int);

        for (int heading = 0; heading < dim_3_size_int; ++heading) {
          dist_heuristic_angles[heading] = dist_heuristic_lookup_table[index];
          indices[heading] = index;
          index ++;
        }
        // we want to get the minimum distance heuristic for all the goal heading angles
        // and that is also one of the goal heading angles
        auto min_it = std::min_element(
          dist_heuristic_angles.begin(), dist_heuristic_angles.end(), [&](float a, float b) {
            int a_index = &a - &dist_heuristic_angles[0];
            int b_index = &b - &dist_heuristic_angles[0];
            return (is_goal_heading(a_index) ? a : std::numeric_limits<float>::max()) <
            (is_goal_heading(b_index) ? b : std::numeric_limits<float>::max());
          });
        // if we find one we update the distance heuristic for all the goal heading angles
        if (min_it != dist_heuristic_angles.end()) {
          for (int heading = 0; heading < dim_3_size_int; ++heading) {
            dist_heuristic_lookup_table[indices[heading]] = *min_it;
          }
        }
      }
    }
  }
}
```